### PR TITLE
Remove Alchemy endpoints, replace with env vars. Add optimism config.

### PR DESCRIPTION
**Pull request by DeusExM4ch1n4**
- Removed Alchemy endpoints, these are used for truffle console tests.
- These are now stored as env vars.
- The optimism goerli dev config was added to the truffle config for console testing.
- ReadMe is updated with Truffle Instructions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,6 +51,77 @@ npm install
...
+    "transactionIndex": 26,
+    "type": "0x0",
+    "rawLogs": [[Object]]
+  },
+  "logs": []
+}
+```
+
+```shell
+truffle(mumbai)> instance.setTokenURI("https://us-central1-share-336018.cloudfunctions.net/share-pfa-0x542896F9FEEBEE3E22C648AA28990C18C08575DC")
```
Comment (un7c0rn) [inline README.md]: @DeusExM4ch1n4 Let's redact the endpoint here.

File: truffle-config.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 const HDWalletProvider = require("@truffle/hdwallet-provider");
 const mnemonicPhrase = process.env.MNEMONIC_PHRASE;
-const RPC_POLYGON_MUMBAI =
-  "https://polygon-mumbai.g.alchemy.com/v2/lcwcK1Go85jK9KMMyIq4PKdzTWucCCCx";
+const RPC_OPTIMISM_GOERLI = process.env.RPC_OPTIMISM_GOERLI;
```
Comment (un7c0rn) [inline truffle-config.js]: nit: s/RPC_/RPC_ENDPOINT_

**Review by un7c0rn — CHANGES_REQUESTED** [review]
Thanks JP! Notes inline.

File: README.md [hunk 3]
```diff
@@ -51,6 +51,77 @@ npm install
...
+From the console you can then call "migrate" which will execute the code in `migrations/1_initial_migration.js` and deploy the respective contracts to the selected blockchain. From the console you can instantiate a reference to the contract using:
+
+```javascript
+let instance = await GNFT.at(
+  "0xF3f95d49D5205d0aD811ebdadA1c106aef7D69f2"
+);
+```
+
+```shell
+truffle(mumbai)> instance.initialize("", "1000000000", 300)
```
Comment (un7c0rn) [inline README.md]: @DeusExM4ch1n4 Why the empty string for the first argument here?

File: README.md [hunk 4]
```diff
@@ -51,6 +51,77 @@ npm install
...
+    "transactionIndex": 26,
+    "type": "0x0",
+    "rawLogs": [[Object]]
+  },
+  "logs": []
+}
+```
+
+```shell
+truffle(mumbai)> instance.setTokenURI("mock/token/uri")
```
Comment (un7c0rn) [inline README.md]: @DeusExM4ch1n4 nano-nit: `/mock/token/uri`

File: README.md (hunk 3 above)
Comment (DeusExM4ch1n4) [reply README.md]: This is a reused example from the old docs in Share ReadMe. It is quite outdated as the pfa initialization accepts many more arguments now. I will update it to a more accurate example.

**Review by un7c0rn — APPROVED** [review]
LGTM

**Pull request merged**